src/dtclient/kubernetes_settings.go

```diff
@@ -231,21 +231,29 @@ func createBaseRequest(url, method, apiToken string, body io.Reader) (*http.Requ
 }
 
 func handleErrorArrayResponseFromAPI(response []byte, statusCode int) error {
-	var se []getSettingsErrorResponse
-	if err := json.Unmarshal(response, &se); err != nil {
-		return fmt.Errorf("response error: %d, can't unmarshal json response: %w", statusCode, err)
-	}
+	if statusCode == http.StatusForbidden || statusCode == http.StatusUnauthorized {
+		var se getSettingsErrorResponse
+		if err := json.Unmarshal(response, &se); err != nil {
+			return fmt.Errorf("response error: %d, can't unmarshal json response: %w", statusCode, err)
```

Comment:
maybe the whole stack trace is unnecessary
just returning the response code and that unmarshal failes should be enough